analyzer/src/main/kotlin/managers/Conan.kt

```diff
@@ -144,25 +142,47 @@ class Conan(
         stashDirectories(directoryToStash).use {
             configureRemoteAuthentication(conanConfig)
 
+            // TODO: Support lockfiles which are located in a different directory than the definition file.
+            val lockfileName = analyzerConfig.packageManagers?.get(managerName)?.options?.get("lockfileName")
+            requireLockfile(workingDir) { lockfileName?.let { hasLockFile(workingDir.resolve(it).path) } ?: false }
```

Comment:
I think we should verify that the provided lockfile is below the `analysisRoot` directory to prevent that users reference files which do not belong to the project. You can see d45a824 for an example where a similar feature was implemented for the `ArchiveUtils` class.

Reply:
@mnonnenmacher Implemented, see `verifyLockfileBelongsToProject` function